src/python/pants/backend/visibility/parser_state.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
not too fond of the file names in the visibility backend... but didn't want to waste time trying to come up with something better right now. can always rename before we land this.